### PR TITLE
Get frost coordinator from random source

### PR DESCRIPTION
**Pull request by scottmillner**
Adds a random source provider that is passed to `SigningStateMachine` and used to determine authority in turn

### CODE REVIEW

File: crates/consensus/authority/src/random_source_provider.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+use reth_revm::primitives::FixedBytes;
+
+/// Trait that provides a random source of 32 bytes.
+pub trait RandomSource: Debug {
+    /// Returns a random source.
+    fn random_source(&self) -> FixedBytes<32>;
+}
+
+/// Struct that provides a random source of 32 bytes.
```
Comment (scottmillner) [inline crates/consensus/authority/src/random_source_provider.rs:12]: assuming 32 bytes which has some logic downstream that expects that but will refactor as needed

File: crates/consensus/common/src/utils.rs [hunk 2]
```diff
@@ -46,51 +49,34 @@ pub fn validate_chain_version(edh_chain_version: u32) -> Result<(), ConsensusErr
...
     signer_index: u64,
-    reference_timestamp: u64,
     block_time: u64,
-) -> CoordinatorInterval {
-    assert!(block_time > 0, "block_time must be greater than 0");
-    // Calculate the length of one complete cycle
-    let cycle_length = authorities_len * block_time;
+    random_source: FixedBytes<32>,
+) -> bool {
+    // convert types to U256 since random_source is 32 bytes and do arithmetic
```
Comment (scottmillner) [inline crates/consensus/common/src/utils.rs:66]: assuming 32 bytes but will refactor as needed

File: crates/consensus/authority/src/random_source_provider.rs (hunk 1 above)
Comment (arminsabouri) [reply crates/consensus/authority/src/random_source_provider.rs:12]: Make generic?

Comment (scottmillner) [reply crates/consensus/authority/src/random_source_provider.rs:12]: via discussion, will assume 32 bytes for now

File: crates/consensus/authority/src/random_source_provider.rs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    /// Creates a new `RandomSourceProvider`.
+    pub fn new() -> Self {
+        Self
+    }
+}
+
+impl RandomSource for RandomSourceProvider {
+    // TODO: Implement a proper random source.
+    fn random_source(&self) -> FixedBytes<32> {
+        FixedBytes::default()
```
Comment (arminsabouri) [inline crates/consensus/authority/src/random_source_provider.rs:26]: what is the default implementation? All 0 bytes?

File: crates/consensus/common/src/utils.rs [hunk 4]
```diff
@@ -46,51 +49,34 @@ pub fn validate_chain_version(edh_chain_version: u32) -> Result<(), ConsensusErr
...
-pub type CoordinatorInterval = (u64, u64, u64, u64);
-
-/// Returns the inturn interval for a signer index based on the seconds passed
-pub fn get_in_turn_interval(
+/// Returns true if the authority is in turn
+pub fn is_inturn(
     authorities_len: u64,
     signer_index: u64,
-    reference_timestamp: u64,
     block_time: u64,
```
Comment (arminsabouri) [inline crates/consensus/common/src/utils.rs]: Would rename to something more generic like period or time_range

**Review by arminsabouri — DISMISSED** [review]

File: crates/consensus/authority/src/random_source_provider.rs (hunk 3 above)
Comment (scottmillner) [reply crates/consensus/authority/src/random_source_provider.rs:26]: yeah `default()` just returns a `const` that's all 0 bytes

File: crates/consensus/common/src/utils.rs (hunk 4 above)
Comment (scottmillner) [reply crates/consensus/common/src/utils.rs]: went with `time_range`

**Review by arminsabouri — APPROVED** [review]

**Pull request merged**